src/reporter.ts

```diff
@@ -0,0 +1,292 @@
+  outputFile: string;
+  shouldUpload: boolean;
+  assets: Asset[];
+  api?: SauceLabs;
+  cucumberVersion: string;
+  startedAt?: Date;
+  endedAt?: Date;
+  videoStartTime?: number;
+  consoleLog: string[];
+  passed: boolean;
+
+  constructor(config: IFormatterOptions) {
+    super(config)
+
+    const reporterConfig: FormatOptions = config.parsedArgvOptions;
+    this.suiteName = reporterConfig?.suiteName || '';
+    this.browserName = reporterConfig?.browserName || 'chrome';
+    this.build = reporterConfig?.build || '';
+    this.tags = reporterConfig?.tags || [];
+    this.region = reporterConfig?.region || 'us-west-1';
+    this.outputFile = reporterConfig?.outputFile || 'sauce-test-report.json';
+    this.shouldUpload = reporterConfig?.upload !== false;
+    this.cucumberVersion = 'unknown'
+    this.testRun = new TestRun();
+    this.assets = [];
+    this.consoleLog = [];
+    this.passed = true;
+
+    let reporterVersion = 'unknown';
+    try {
+      const packageData = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'package.json'), 'utf-8'));
+      reporterVersion = packageData.version;
+    // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    if (process.env.SAUCE_USERNAME && process.env.SAUCE_USERNAME !== '' && process.env.SAUCE_ACCESS_KEY && process.env.SAUCE_ACCESS_KEY !== '') {
+      this.api = new SauceLabs({
+        user: process.env.SAUCE_USERNAME,
+        key: process.env.SAUCE_ACCESS_KEY,
+        region: this.region,
+        headers: {
+          'User-Agent': `cucumber-reporter/${reporterVersion}`
+        },
+      });
+      this.api.tld = this.region === 'staging' ? 'net' : 'com';
+    }
+    if (process.env.SAUCE_VIDEO_START_TIME) {
+      this.videoStartTime = new Date(process.env.SAUCE_VIDEO_START_TIME).getTime();
+    }
+
+    config.eventBroadcaster?.on('envelope', (envelope: Envelope) => {
+      if (envelope.testCaseFinished) {
+        this.logTestCase(envelope.testCaseFinished)
+      }
+      if (envelope.testRunFinished) {
+        this.logTestRun(envelope.testRunFinished)
+      }
+    })
+  }
+
+  logTestCase(testCaseFinished: { testCaseStartedId: any }) {
+    const testCaseAttempt = this.eventDataCollector.getTestCaseAttempt(testCaseFinished.testCaseStartedId)
+    const parsed = formatterHelpers.parseTestCaseAttempt({
+      //cwd: this.cwd,
+      snippetBuilder: this.snippetBuilder, 
+      supportCodeLibrary: this.supportCodeLibrary,
+      testCaseAttempt 
+    })
+    const suite = new Suite(parsed.testCase?.sourceLocation?.uri || '')
+    const curr = new Suite(parsed.testCase?.name);
+    const suiteStatus = parsed.testCase?.worstTestStepResult?.status?.toLowerCase();
+    curr.status = suiteStatus as SauceStatus;
+    curr.metadata = {
+      attempt: parsed.testCase?.attempt,
+      sourceLocation: parsed.testCase?.sourceLocation,
+    }
+    this.consoleLog.push(`${curr.name}\t#${suite.name}`)
+
+    parsed.testSteps.forEach((testStep: IParsedTestStep) => {
```

Comment:
You don't need an explicit type for `testStep`. The `testSteps` property is a `IParsedTestStep[]` type so the iterator has an inferred type. Removing this also removes the need to import a private type (https://github.com/saucelabs/sauce-cucumber-reporter/blob/ed7ac7ffdc0ad1bcedef34f72419881a2cb48989/src/reporter.ts#L7)